### PR TITLE
Use staff-sso for django admin login

### PR DESCRIPTION
**Pull request by kerin**
`django-staff-sso-client` integrates with `staff-sso` and automatically creates a django `User` object on initial login; it does not however set `is_staff` to `True`, so at the moment that needs to be modified manually.

Django admin's username/password login is replaced with OAuth2 by changing the `/admin/login/` view to redirect to the oauth2 login flow - there may be a simpler/cleaner approach in the long term.

### CODE REVIEW

File: server/settings/components/common.py [hunk 1]
```diff
@@ -175,6 +177,7 @@
 AUTHENTICATION_BACKENDS = (
     'axes.backends.AxesBackend',
     'django.contrib.auth.backends.ModelBackend',
```
Comment (r4vi) [inline server/settings/components/common.py]: can we delete ModelBackend?

File: server/settings/components/common.py [hunk 2]
```diff
@@ -205,3 +208,13 @@
 
 # Timeouts
 EMAIL_TIMEOUT = 5
+
+# Authbroker
+AUTHBROKER_URL = "https://sso.trade.gov.uk/"
+AUTHBROKER_CLIENT_ID = "B9xGM4WWRuQ2eYTALaV3RTS8AV4beSyTnTSk6tO1"
```
Comment (r4vi) [inline server/settings/components/common.py]: hope these are fake

File: server/urls.py [hunk 3]
```diff
@@ -31,8 +31,12 @@
     # Health checks:
     path('health/', include(health_urls)),  # noqa: DJ05
 
+    # Django Admin Oauth2
+    path("auth/", include("authbroker_client.urls")),
+
     # django-admin:
     path('admin/doc/', include(admindocs_urls)),  # noqa: DJ05
+    path('admin/login/', RedirectView.as_view(pattern_name="authbroker_client:login", permanent=False)),
```
Comment (r4vi) [inline server/urls.py]: seems fine to me, to do it this way

File: server/settings/components/common.py [hunk 4]
```diff
@@ -205,3 +208,13 @@
...
+AUTHBROKER_CLIENT_SECRET = "d5Yv2YijSSMj0VNxUZ570JEjVpKh5yzqmkHcWixALz5MsR0seCwn2GP1DO3fPqbSma1jBpuIkHdKeIlKHGk5PFDb6D6qU1uWDlgu5r0sfQJtTEw4BEkonTOuFWdDJayG"
```
Comment (r4vi) [inline server/settings/components/common.py]:
```suggestion
AUTHBROKER_CLIENT_SECRET = config('AUTHBROKER_CLIENT_SECRET')
```
probably best to access indirectly through `config` and set in .env

Comment (kerin) [reply server/settings/components/common.py]: 🤦‍♂

File: server/settings/components/common.py (hunk 1 above)
Comment (kerin) [reply server/settings/components/common.py]: Not entirely sure at the minute - if SSO users are automatically created with `is_staff=False` we might want to be able to log in with username/password to make those SSO users staff via the admin UI 🤷‍♂

**Comment by kerin** [comment]
@r4vi unfortunate slipup addressed and blind redirect replaced with a login link in place of the django admin login form.

**Pull request merged**